recursion/writing-recursive-code.md

```diff
@@ -0,0 +1,183 @@
+# Writing Recursive Code
+
+## Learning Goals
+
+- Construct a recursive solution from a word problem
+
+## Considerations for a Recursive Function
+
+When writing a recursive function, the following prompts will help construct different parts of a recursive function.
+
+- What is the base case? What is the version of the problem where we know the answer without recursing?
+- How can we describe the repeated problem that gets divided into smaller versions of the problem?
+- What information does each smaller problem need?
+- What information do we want back from each problem?
+- How do we expect examples to behave?
+
+### Determine the Base Case
+
+The base case is the the version of the problem where we can determine the answer without recursing. Generally, the base case is used as the condition which ends the recursion.
+
+We may be able to determine the base case from the given problem statement, or we may need to ask clarifying questions.
```

Comment:
provide some other alternatives for how we might deduce this. it's rarely an exact science
```suggestion
We may be able to determine the base case directly from the given problem statement. Sometimes we may have additional understanding that lets us determine the base case. Other times we may need to ask clarifying questions, or experiment with a few sample situations to get a feel for what might be a workable base case.
```